src/main/java/uk/gov/hmcts/probate/schedule/SendNotificationsTask.java

```
@@ -0,0 +1,81 @@
+    private final DataExtractDateValidator dataExtractDateValidator;
+    private final AutomatedNotificationService automatedNotificationService;
+    private final FeatureToggleService featureToggleService;
+
+
+    @Value("${automated_notification.stop_reminder.first_notification_days}")
+    private int firstNotificationDays;
+
+    @Value("${automated_notification.stop_reminder.second_notification_days}")
+    private int secondNotificationDays;
+
+    @Value("${adhocSchedulerJobDate}")
+    public String adHocJobDate;
+
+    @Override
+    public void run() {
+        log.info("Scheduled task SendNotificationsTask started");
+        String firstStopReminderDate = DATE_FORMAT.format(LocalDate.now().minusDays(firstNotificationDays));
+        String secondStopReminderDate = DATE_FORMAT.format(LocalDate.now().minusDays(secondNotificationDays));
+        if (StringUtils.isNotEmpty(adHocJobDate)) {
+            log.info("Running SendNotificationsTask with Adhoc dates {}", adHocJobDate);
+            firstStopReminderDate = LocalDate.parse(adHocJobDate).minusDays(firstNotificationDays).toString();
+            secondStopReminderDate = LocalDate.parse(adHocJobDate).minusDays(secondNotificationDays).toString();
+        }
+
+        try {
+            if (!featureToggleService.isFirstStopReminderFeatureToggleOn()) {
+                log.info("Feature toggle FirstStopReminderFeatureToggle is off, skipping task");
+            } else {
+                log.info("Calling Send First Stop Reminder for date {}", firstStopReminderDate);
+                dataExtractDateValidator.dateValidator(firstStopReminderDate);
+                log.info("Perform Send First Stop Reminder started");
+                automatedNotificationService.sendNotification(firstStopReminderDate, FIRST_STOP_REMINDER);
+                log.info("Perform Send First Stop Reminder finished");
+
+            }
+        } catch (ApiClientException e) {
+            log.error(e.getMessage());
```

do we really just want to log the message (and not say the stack trace so we can work out what's happened)? similar question for the other catch & log instances - some of them may be reasonable, but if we're totally discarding the exception maybe not ideal.

this is especially important for the instance immediately below because that could be any Exception subclass, which might be signalling a failure we did not anticipate at all